### ISSUE TITLE
coordinate eltype

### ISSUE THREAD

**Issue by visr** [JuliaGeo/GeoInterface.jl#53]
Do we currently expect anything from the coordinate eltype? Was SF always Float64? What do we do, `<:Real`, `<:AbstractFloat`, or leave it open?

**Comment by evetion** [comment]
This is already in the documentation itself:
`GeoInterface.getcoord(geomtype(geom), geom::customgeom, i)::Real`

I clarified it in the docstring as well:
```
    getcoord(geom, i) -> Number
Return the `i`th coordinate for a given `geom`. A coordinate isa *`Real`*.
Note that this is only valid for individual [`AbstractPointTrait`](@ref)s.
```

**Comment by visr** [comment]
Yeah indeed I think just documenting it might be enough, rather than "enforcing". Of course some operations may only make sense on AbstractFloat coordinates, but that is up to the implementer.

**Comment by evetion** [comment]
@rafaqz Brought up the point to change `Real` into `Number` so things like Unitful.jl could work. This wouldn't be a breaking change.

**Comment by asinghvi17** [comment]
https://github.com/JuliaGeo/GeoInterface.jl/issues/128
